openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -896,10 +916,122 @@ def test_learner_eligibility_for_one_click_purchase(self):
         data = ProgramMarketingDataExtender(program, self.user).extend()
         self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
 
-        courses.append(self._create_course(self.course_price, is_enrolled=True))
+        course = self._create_course(self.course_price)
+        CourseEnrollmentFactory(user=self.user, course_id=course['course_runs'][0]['key'])
         program2 = ProgramFactory(
-            courses=courses,
+            courses=[course],
             is_program_eligible_for_one_click_purchase=True
         )
         data = ProgramMarketingDataExtender(program2, self.user).extend()
         self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
+
+    def test_multiple_published_course_runs(self):
+        """
+        Learner should not be eligible for one click purchase if:
+            - program has a course with more than one published course run
+        """
+        course_run_1 = CourseRunFactory(
+            key=str(ModuleStoreCourseFactory().id),
+            status='published'
+        )
+        course_run_2 = CourseRunFactory(
+            key=str(ModuleStoreCourseFactory().id),
+            status='published'
+        )
+        course = CourseFactory(course_runs=[course_run_1, course_run_2])
+        program = ProgramFactory(
+            courses=[course],
+            is_program_eligible_for_one_click_purchase=True
+        )
+        data = ProgramMarketingDataExtender(program, self.user).extend()
+
+        self.assertFalse(data['is_learner_eligible_for_one_click_purchase'])
+
+        course_run_2['status'] = 'unpublished'
```

Comment:
Is `course_run_2` an object or a dictionary? Since it's created using a factory I'm guessing an object, so the dot notation might be more natural to use here.